src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java

```diff
@@ -225,8 +250,11 @@ private void addHomelessCitizens()
             // Move the citizen to a better hut
             if (citizen.getHomeBuilding() != null && citizen.getHomeBuilding().getBuildingLevel() < this.getBuildingLevel())
             {
-                // The citizen can move to this hut to improve conditions
-                citizen.getHomeBuilding().removeCitizen(citizen);
+                if(citizen.getHomeBuilding() instanceof BuildingHome)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1066)